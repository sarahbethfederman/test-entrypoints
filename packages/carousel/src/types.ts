import { LUIGlobalProps } from '@lendi-ui/utils';
import { CSSProperties, ReactElement, FunctionComponent } from 'react';

export interface CarouselSession {
  isNextDisabled: boolean;
  isPreviousDisabled: boolean;
  nextSlide: () => void;
  previousSlide: () => void;
  setSlide: (newSlide: number) => void;
  slides: number;
  currentIndex: number;
  increment: number;
  infinite: boolean;
  containerBox: ClientRect | DOMRect;
}

export interface SlidesProps extends LUIGlobalProps {
  setSlides: (slides: number) => void;
  height: string;
  speed: number;
  windowWidth: number;
}

export interface SlidesState {
  width: number;
  currentIndex: number;
  previousIndex: number;
  slideWidths: number[];
}

export interface IncrementMap {
  mobile: number;
  tablet: number;
  desktop: number;
}

export interface CarouselProps extends LUIGlobalProps {
  width?: string;
  height?: string;
  initialIndex?: number;
  increment?: number | IncrementMap;
  withArrows?: boolean;
  withDots?: boolean;
  infinite?: boolean;
  autoplay?: boolean;
  duration?: number;
  speed?: number;
  swipe?: boolean;
}

export interface CarouselArrowsProps extends LUIGlobalProps {
  width?: number;
  children?: ((disabled: boolean) => ReactElement) | ReactElement;
}

export interface CarouselDotsProps extends LUIGlobalProps {
  children?: ((isActive: boolean) => ReactElement) | ReactElement;
  style?: CSSProperties;
}

export interface CarouselArrowOffset {
  x: number;
  y: number;
}

export interface CarouselArrowStyleProps {
  disabled?: boolean;
  offset?: CarouselArrowOffset;
  hide?: boolean;
}

export interface CarouselDotsStyleProps {
  isActive?: boolean;
}

export interface CarouselCompoundComponent {
  Slides: FunctionComponent;
  NextArrow: FunctionComponent;
  PreviousArrow: FunctionComponent;
  Dots: FunctionComponent<CarouselDotsProps>;
}
