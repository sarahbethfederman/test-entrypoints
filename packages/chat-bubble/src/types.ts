import { LUIGlobalProps } from '@lendi-ui/utils';

export type Direction = 'left' | 'right';

export interface ChatBubbleContentProps extends LUIGlobalProps {}

export interface ChatBubbleFooterProps extends LUIGlobalProps {}

export interface SidebarCompoundComponent {
  Content: React.FunctionComponent<ChatBubbleContentProps>;
  Footer: React.FunctionComponent<ChatBubbleFooterProps>;
}

export interface ChatBubbleProps extends LUIGlobalProps {
  direction?: Direction;
}
