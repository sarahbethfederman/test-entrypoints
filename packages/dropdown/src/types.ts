import { LUIGlobalProps } from '@lendi-ui/utils';
import { BreakpointValueMap, BreakpointValue } from '@lendi-ui/breakpoint';
import { ThemeMap } from '@lendi-ui/theme';

export interface OptionType {
  [key: string]: any;
}
export type OptionsType<OptionType> = ReadonlyArray<OptionType>;
export type ValueType<OptionType> = OptionType | OptionsType<OptionType> | null | undefined;

export type Size = 'xs' | 'sm' | 'md' | 'lg';
export type InputSize = BreakpointValue<Size> | BreakpointValueMap<Size>;

export type FocusEventHandler = (event: React.FocusEvent<HTMLElement>) => void;
export type KeyboardEventHandler = (event: React.KeyboardEvent<HTMLElement>) => void;
export type FormatOptionLabelContext = 'menu' | 'value';

export interface LUISelectProps extends LUIGlobalProps {
  className?: string;
  /* Formats option labels in the menu and control as React components so that consumer can show menu-items with before and after icon or whatever way he wants to
  use it - see examples - `select-before-icon-example` and `select-with-description-example` */
  formatOptionLabel?: (option: OptionType, { context }: { context: FormatOptionLabelContext }) => React.ReactNode;
  /* Hide the selected option from the menu */
  hideSelectedOptions?: boolean;
  /* The value of the search input */
  inputValue?: string;
  isAutoFocus?: boolean;
  /* Is the select value clearable */
  isClearable?: boolean;
  /* Remove the currently focused option when the user presses backspace */
  isClearableByBackspace?: boolean;
  isDisabled?: boolean;
  isError?: boolean;
  isFullWidth?: boolean;
  isInverse?: boolean;
  isMenuOpen?: boolean;
  isOptionsOverflow?: boolean;
  /* enable multi select*/
  isMultiple?: boolean;
  isSearchable?: boolean;
  onBlur?: FocusEventHandler;
  onChangeItem?: (value: ValueType<OptionType>) => void;
  onFocus?: FocusEventHandler;
  onKeyDown?: KeyboardEventHandler;
  onInputChange?: (val: string) => string | void;
  onMenuOpen?: () => void;
  onMenuClose?: () => void;
  name?: string;
  /* Array of options that populate the select menu  - it is required*/
  options: OptionsType<OptionType>;
  placeholder?: string;
  size?: Size;
  theme?: ThemeMap;
  /* The value of the select; reflected by the selected option */
  value?: ValueType<OptionType>;
}
