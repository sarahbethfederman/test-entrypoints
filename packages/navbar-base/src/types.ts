import { FunctionComponent, ReactNode } from 'react';
import { LUIGlobalProps } from '@lendi-ui/utils';

export interface NavbarBaseProps extends LUIGlobalProps {
  isTransparent?: boolean;
  className?: string;
  children?: ReactNode;
}
export interface NavbarCompoundComponent {
  Left: FunctionComponent;
  Right: FunctionComponent;
}
export interface NavbarBaseSession {
  isTransparent?: boolean;
}

export interface NavbarBaseLeftProps extends LUIGlobalProps {
  children?: ReactNode;
}

export interface NavbarBaseRightProps extends LUIGlobalProps {
  children?: ReactNode;
}
