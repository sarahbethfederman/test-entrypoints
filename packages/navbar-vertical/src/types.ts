import { FunctionComponent } from 'react';
import { LUIGlobalProps } from '@lendi-ui/utils';

export interface VerticalNavbarProps extends LUIGlobalProps {}

export interface StyledVerticalNavbarProps {
  expanded: boolean;
}

export interface VerticalNavState {
  isMouseOver: boolean;
  isExpanded: boolean;
  setIsExpanded: () => void;
}

export interface VerticalNavbarCompoundComponent {
  Header: FunctionComponent<VerticalNavHeaderProps>;
  Tab: FunctionComponent<VerticalNavTabProps>;
}

export interface VerticalNavHeaderProps extends LUIGlobalProps {}

export interface VerticalNavTabProps extends LUIGlobalProps {
  selected?: boolean;
  tooltip?: string;
  onClick?: () => void;
  maxHeight?: string;
  disabled?: boolean;
}
