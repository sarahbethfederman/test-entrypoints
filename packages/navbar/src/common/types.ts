export enum ApplicationStage {
  started_application = 'started_application',
  finished_funnel2 = 'finished_funnel2',
  finished_funnel3 = 'finished_funnel3',
  finalised_application = 'finalised_application',
  loan_selected = 'loan_selected',
}

export type ApplicationType = 'Refinance' | 'New Purchase';

export interface Applicant {
  uuid?: string;
  sfid?: string;
  type?: 'primary' | string;
  firstName?: string;
}
export interface Application {
  type?: ApplicationType;
  sfid?: string;
  number?: number;
  stage?: ApplicationStage;
  createdDate?: string;
  continueURL?: string;
  applicants?: Applicant[];
  isAuth?: boolean;
}
export interface Broker {
  fullName: string;
  photo: string;
  phone: string;
  title: string;
}

export interface ApplicationInSF {
  activeApplicationNumber?: number;
  activeApplicationSfid?: string;
  applicants?: ApplicantInSF[];
  applicationStage?: string;
  applicationCreatedDate?: string;
  continueApplicationLink?: string;
  transactionType?: string;
  isAuth?: boolean;
}

export interface ApplicantInSF {
  applicant_type?: string;
  first_name?: string;
  married_to_uuid?: any;
  sfid?: string;
  uuid?: string;
}
