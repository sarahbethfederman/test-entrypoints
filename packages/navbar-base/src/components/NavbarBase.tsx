import * as React from 'react';
import NavbarLeft from './NavbarLeft';
import NavbarRight from './NavbarRight';
import * as Sticky from 'react-stickynode';
import { NavbarBaseProps, NavbarCompoundComponent } from '../types';
import { ChildrenWrapper } from '../index.style';
import NavbarBaseContext from '../context/NavbarContext';

const NavbarBase: React.FunctionComponent<NavbarBaseProps> & NavbarCompoundComponent = ({
  children,
  isTransparent: useTransparent = false,
  className,
}) => {
  const [isTransparent, setIsTransparent] = React.useState<boolean>(useTransparent);

  React.useEffect(() => {
    setIsTransparent(useTransparent);
  }, [useTransparent]);

  const handleStickyState = ({ status }: Sticky.Status) => {
    if (useTransparent) {
      if (status === Sticky.STATUS_FIXED) {
        setIsTransparent(false);
      } else {
        setIsTransparent(true);
      }
    }
  };
  return (
    <nav aria-label="Main navigation">
      <NavbarBaseContext.Provider value={{ isTransparent }}>
        <Sticky innerZ={3} onStateChange={handleStickyState}>
          <ChildrenWrapper isTransparent={isTransparent} className={className}>
            {children}
          </ChildrenWrapper>
        </Sticky>
      </NavbarBaseContext.Provider>
    </nav>
  );
};

NavbarBase.Left = NavbarLeft;
NavbarBase.Right = NavbarRight;

export default NavbarBase;
