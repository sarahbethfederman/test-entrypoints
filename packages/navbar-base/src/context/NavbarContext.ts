import { createContext, useContext } from 'react';
import { NavbarBaseSession } from '../types';

const NavbarBaseContext = createContext<NavbarBaseSession | undefined>(undefined);

export const useNavbarBaseContext = () => {
  const context = useContext(NavbarBaseContext);

  if (!context) {
    throw new Error(`NavbarBase compound components cannot be rendered outside the NavbarBase component`);
  }

  return context;
};

export default NavbarBaseContext;
