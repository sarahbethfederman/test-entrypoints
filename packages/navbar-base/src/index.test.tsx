import * as React from 'react';
import { mount, ReactWrapper } from 'enzyme';
import Theme from '@lendi-ui/theme';
import NavbarBase from './components/NavbarBase';
import { NavbarBaseProps } from './types';
import { useNavbarBaseContext } from './context/NavbarContext';
import { Button, ButtonGroup } from '@lendi-ui/button';
import styled from 'styled-components';
import { ChildrenWrapper } from './index.style';

let element: ReactWrapper<NavbarBaseProps>;

describe('NavbarBase', () => {
  beforeEach(() => {
    element = mount(
      <Theme>
        <NavbarBase />
      </Theme>
    );
  });

  it('should mount', () => {
    expect(element.find(NavbarBase)).toHaveLength(1);
    expect(element.find(NavbarBase)).toMatchSnapshot();
  });
});

describe('Navbarbase with children', () => {
  beforeEach(() => {
    element = mount(
      <Theme>
        <NavbarBase>
          <NavbarBase.Left>
            <a href="/">The link</a>
          </NavbarBase.Left>
          <NavbarBase.Right>
            <button>The button</button>
          </NavbarBase.Right>
        </NavbarBase>
      </Theme>
    );
  });
  it('should render with Left, Center and Right content', () => {
    expect(element.find('a').text()).toEqual('The link');
    expect(element.find('button').text()).toEqual('The button');
  });
});

const InitButtons = () => {
  const { isTransparent } = useNavbarBaseContext();
  return (
    <ButtonGroup size="sm" isInverse={isTransparent}>
      <ButtonGroup.Button variant="secondary">Continue</ButtonGroup.Button>
      <ButtonGroup.Button variant="secondary">Next</ButtonGroup.Button>
    </ButtonGroup>
  );
};

describe('Navbarbase children to use Transparent property from NavbarBaseContext', () => {
  beforeEach(() => {
    element = mount(
      <Theme>
        <NavbarBase isTransparent={true}>
          <NavbarBase.Left>
            <a href="/">The link</a>
          </NavbarBase.Left>
          <NavbarBase.Right>
            <InitButtons />
          </NavbarBase.Right>
        </NavbarBase>
      </Theme>
    );
  });
  it('should render with inverse styled button', () => {
    const buttons = element.find(Button);
    buttons.forEach((button) => expect(button.prop('isInverse')).toBeTruthy());
  });
});

describe('Navbarbase to entertain style from consumer', () => {
  const NavbarBaseWrapper = styled(NavbarBase)`
    height: 100px;
  `;
  beforeEach(() => {
    element = mount(
      <Theme>
        <NavbarBaseWrapper isTransparent={true}>
          <NavbarBase.Left>
            <a href="/">The link</a>
          </NavbarBase.Left>
          <NavbarBase.Right>
            <InitButtons />
          </NavbarBase.Right>
        </NavbarBaseWrapper>
      </Theme>
    );
  });
  it('should render with inverse styled button', () => {
    const childrenWrapper = element.find(ChildrenWrapper);
    expect(childrenWrapper).toHaveStyleRule('height', '100px');
  });
});
