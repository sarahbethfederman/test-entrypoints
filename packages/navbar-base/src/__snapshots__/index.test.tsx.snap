// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavbarBase should mount 1`] = `
.c0 {
  box-sizing: border-box;
  line-height: 1.333;
  white-space: normal;
  border-spacing: 2px;
  text-align: left;
  word-spacing: normal;
  -webkit-letter-spacing: normal;
  -moz-letter-spacing: normal;
  -ms-letter-spacing: normal;
  letter-spacing: normal;
  padding: 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  width: 100%;
  box-shadow: 0 3px 3px 0 rgba(0,0,0,0.1), 0 3px 4px 0 rgba(0,0,0,0.12), 0 1px 8px 0 rgba(0,0,0,0.14);
  background-color: rgba(255,255,255,1);
}

<NavbarBase>
  <nav
    aria-label="Main navigation"
  >
    <Sticky
      activeClass="active"
      bottomBoundary={0}
      enableTransforms={true}
      enabled={true}
      innerClass=""
      innerZ={3}
      onStateChange={[Function]}
      releasedClass="released"
      shouldFreeze={[Function]}
      top={0}
    >
      <div
        className="sticky-outer-wrapper"
        style={Object {}}
      >
        <div
          className="sticky-inner-wrapper "
          style={
            Object {
              "position": "relative",
              "top": "",
              "transform": "translate3d(0,0px,0)",
              "zIndex": 3,
            }
          }
        >
          <styled.div
            isTransparent={false}
          >
            <StyledComponent
              forwardedComponent={
                Object {
                  "$$typeof": Symbol(react.forward_ref),
                  "attrs": Array [],
                  "componentStyle": ComponentStyle {
                    "componentId": "sc-bwzfXH",
                    "isStatic": false,
                    "lastClassName": "c0",
                    "rules": Array [
                      "
  ",
                      "
  box-sizing: border-box;
  line-height: 1.333;
  white-space: normal;
  border-spacing: 2px;
  text-align: left;
  word-spacing: normal;
  letter-spacing: normal;
",
                      ";
  padding: 0;
  display: flex;
  flex-direction: row;
  align-items: center;
  width: 100%;

  ",
                      [Function],
                      "
",
                    ],
                  },
                  "displayName": "styled.div",
                  "foldedComponentIds": Array [],
                  "render": [Function],
                  "styledComponentId": "sc-bwzfXH",
                  "target": "div",
                  "toString": [Function],
                  "warnTooManyClasses": [Function],
                  "withComponent": [Function],
                }
              }
              forwardedRef={null}
              isTransparent={false}
            >
              <div
                className="c0"
              />
            </StyledComponent>
          </styled.div>
        </div>
      </div>
    </Sticky>
  </nav>
</NavbarBase>
`;
